tests/runtime/engine/utils.py

```diff
@@ -99,6 +100,17 @@ def run_test(test):
 
             if test.before:
                 before = subprocess.Popen(test.before, shell=True, preexec_fn=os.setsid)
+                waited=0
+                with open(devnull, 'w') as dn:
+                    # This might not work for complicated cases, such as if
+                    # a test program needs to accept arguments. It covers the
+                    # current simple calls with no arguments
+                    child_name = os.path.basename(test.before.split()[-1])
+                    while subprocess.call("pidof %s" % child_name, shell=True, stdout=dn, stderr=dn) != 0:
```

Comment:
won't this work without shell? Less stuff to spawn

Reply:
@fbs you're right, done! I've been burned by calls like this not passing PATH properly but seems python is ok.